api/controllers/CheckUploadedDocumentsController.js

```diff
@@ -14,20 +17,30 @@ const CheckUploadedDocumentsController = {
             if (uploadedFileData.length === 0) {
                 throw new Error('No files uploaded');
             }
-            uploadedFileData.forEach(async (uploadedFile) => {
-                await UploadedDocumentUrls.create(
+            uploadedFileData.forEach((uploadedFile) => {
+                UploadedDocumentUrls.create(
                     CheckUploadedDocumentsController._dbColumnData(
                         uploadedFile,
                         req
                     )
-                );
-                sails.log.info(
-                    `Url for document ${uploadedFile.filename} added to db`
-                );
+                )
+                    .then(() => {
+                        sails.log.info(
+                            `Url for document ${uploadedFile.filename} added to db`
+                        );
+                    })
+                    .catch((err) => {
+                        sails.log.error(err);
+                        res.serverError();
+                    });
```

Comment:
Appreciate you can't use async/await in a forEach callback, but this code is still asynchronous so will run in parallel to the rest of the function. If there is an error it's likely that `CheckUploadedDocumentsController._checkDocumentCountAndPaymentDetails` will already have been executed and `res.serverError();` would then fail (I'm assuming a few things here.. :) ) 

You've got a couple of options - chain the whole thing in promises or use a classic for loop with await (though this would require `renderPage` to be async, not sure if that's supported.

Reply:
Ah yeah, this is a good point. I like the idea of chaining the whole thing in promises but I don't want to affect `renderPage`  🤔  I will look into it.]

- [ ] Execute `_checkDocumentCountAndPaymentDetails ` only after forEach is complete